tests/integration/helpers.py

```diff
@@ -2,22 +2,152 @@
 # Copyright 2024 Canonical Ltd.
 # See LICENSE file for licensing details.
 
+import logging
 import time
 from pathlib import Path
-from typing import List
+from typing import List, Optional
 
 import yaml
+from juju.application import Application
+from juju.errors import JujuError
+from juju.model import Model
 from juju.unit import Unit
 from pytest_operator.plugin import OpsTest
+from vault import Vault
 
 # Vault status codes, see
 # https://developer.hashicorp.com/vault/api-docs/system/health for more details
 METADATA = yaml.safe_load(Path("./charmcraft.yaml").read_text())
 APP_NAME = METADATA["name"]
-VAULT_STATUS_ACTIVE = 200
-VAULT_STATUS_UNSEALED_AND_STANDBY = 429
-VAULT_STATUS_NOT_INITIALIZED = 501
-VAULT_STATUS_SEALED = 503
+SELF_SIGNED_CERTIFICATES_APPLICATION_NAME = "self-signed-certificates"
+VAULT_PKI_REQUIRER_APPLICATION_NAME = "tls-certificates-requirer"
+
+
+def get_app(model: Model, app_name: str = APP_NAME) -> Application:
+    """Get the application by name.
+
+    Abstracts some of the boilerplate code needed to get the application by
+    caused by the type stubs being non-committal.
```

Comment:
I think an edit left this sentence in a nonsensical state.

Reply:
https://github.com/canonical/vault-operator/pull/271/commits/565660b5dff104b5c9102ef40dec5fd2f6569e1d